### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by JeenaYun**
Hi, I updated the documentation of the **'My first model'** section. A summary of things added are as follows:

- **Mesh creation with Gmsh**: added lines about curvilinear meshing
- **Lua scripting**: added a table that describes material/frictional parameters and their formats (constant/functional), units, and default values. Added comments to the example lua script and created `examples/tandem/2d/tutorial.lua` based on the given example.
- **Parameter file**: added descriptions about parameters including their possible options and default values. Added comments to the example toml script and created `examples/tandem/2d/tutorial.toml` based on the given example.
- **Run model**: added better explanation about the syntax and usage of PETSc options.
- **Post processing**: added a table that describes fault and domain outputs and their units.

Please comment here if you find any misleading descriptions or missing information.

### CODE REVIEW

File: docs/first-model/lua-scripting.rst [hunk 1]
```diff
@@ -1,33 +1,79 @@
 Lua scripting
 =============
 
-.. warning::
-
-   This page is under construction.
+Material and frictional parameters used in Tandem simulation are written in Lua script. Here's key parameters and their descriptions:
```
Comment (NicoSchlw) [inline docs/first-model/lua-scripting.rst:4]: Material and frictional parameters used in Tandem simulation**s** are written in Lua script. **Here are the** key parameters and their descriptions:

File: docs/first-model/parameters.rst [hunk 2]
```diff
@@ -1,20 +1,66 @@
 Parameter file
 ==============
 
-.. warning::
+Tandem simulation parameters are written in toml script. Here's key parameters and their descriptions:
```
Comment (NicoSchlw) [inline docs/first-model/parameters.rst:4]: **Here are the** key parameters and their descriptions

File: docs/first-model/parameters.rst [hunk 3]
```diff
@@ -1,20 +1,66 @@
...
+- **mode**: Mode of SEAS simulation. Available options: [QuasiDynamic/QD | QuasiDynamicDiscreteGreen/QDGreen | FullyDynamic/FD]
+- **ref_normal**: Define reference normal vector.
+- **up**: Define up direction vector.
+- **boundary_linear**: Assert that boundary is a linear function of time (i.e. boundary(x, t) = f(x) t). Default = false
+- **matrix_free**: Use matrix-free operators. Default = false.
+- **cfl**: CFL tuning parameter (typically >= 1.0). Default = 1.0.
+- **mg_coarse_level**: Polynomial degree of coarsest MG level. Default = 1.
+- **mg_strategy**: MG level selection strategy. Available options: [TwoLevel | Logarithmic | Full]. Default = TwoLevel
+
+When using :code:`mode=QDGreen`, you can use Green's function checkpointing feature by defining following parameters:
```
Comment (NicoSchlw) [inline docs/first-model/parameters.rst:20]: you can use Green's function checkpointing by defining **the** following parameters:

File: docs/first-model/parameters.rst [hunk 4]
```diff
@@ -1,20 +1,66 @@
...
-   lib = "tutorial.lua"
-   scenario = "normal"
-   type = "elasticity"
-   ref_normal = [1, 0]
-   boundary_linear = true
+   final_time = 47304000000      # Final simulation time in seconds.
+   mesh_file = "tutorial.msh"    # Mesh file.
+   lib = "tutorial.lua"          # Lua file containing material&frictional paramters.
+   scenario = "normal"           # Name of the specific scenario defined in the lua library.
+   type = "elasticity"           # Type of problems: poisson, elastic/elasticity.
```
Comment (NicoSchlw) [inline docs/first-model/parameters.rst:59]: Type of **problem**: poisson, elastic/elasticity.

File: docs/first-model/run-model.rst [hunk 5]
```diff
@@ -1,11 +1,25 @@
 Run model
 =========
 
-.. warning::
+Tandem simulation can be run with a syntax of :code:`$ tandem parameter_file.toml {tandem options} --petsc {PETSc options}`.
```
Comment (NicoSchlw) [inline docs/first-model/run-model.rst:4]: Tandem simulation**s** can be run with a syntax of

**Review by NicoSchlw — COMMENTED** [review]
Hi Jeena, thanks a lot for this PR, it's very useful! I just added a few grammatical suggestions.

File: docs/first-model/lua-scripting.rst (hunk 1 above)
Comment (Thomas-Ulrich) [reply docs/first-model/lua-scripting.rst:4]: overall, I would just copy paste the file to grammarly, and try to fix the typos.

File: docs/first-model/parameters.rst [hunk 6]
```diff
@@ -1,20 +1,66 @@
...
+- **mesh_file**: Mesh file.
+- **lib**: Lua file containing material&frictional paramters.
+- **scenario**: Name of the specific scenario defined in the lua library.
+- **type**: Type of problems. Available options: [poisson | elastic/elasticity].
+- **mode**: Mode of SEAS simulation. Available options: [QuasiDynamic/QD | QuasiDynamicDiscreteGreen/QDGreen | FullyDynamic/FD]
+- **ref_normal**: Define reference normal vector.
+- **up**: Define up direction vector.
+- **boundary_linear**: Assert that boundary is a linear function of time (i.e. boundary(x, t) = f(x) t). Default = false
+- **matrix_free**: Use matrix-free operators. Default = false.
+- **cfl**: CFL tuning parameter (typically >= 1.0). Default = 1.0.
```
Comment (Thomas-Ulrich) [inline docs/first-model/parameters.rst:16]: typically <= 1.0

File: docs/first-model/lua-scripting.rst [hunk 7]
```diff
@@ -1,33 +1,79 @@
...
++--------------------+---------------------------+-----------------------------------------------------------------+---------+---------+
+| **source**         | Functional (space, time)  | (Optional) Defines an additional source term :math:`s` to the   |         |         |
+|                    |                           | state evolution equation :math:`\psi_{,t} = G(V,\psi) + s(x,t)` |         |         |
++--------------------+---------------------------+-----------------------------------------------------------------+---------+---------+
+| **fault_solution** | Functional (space, time)  | (Optional) Defines slip and state variable. Useful for          |         |         |
+|                    |                           | analytic solutions or manufactured solutions.                   |         |         |
++--------------------+---------------------------+-----------------------------------------------------------------+---------+---------+
+
+For more descriptions about units and scaling, please check out :doc:`Equation scaling section <../reference/scaling>`.
```
Comment (Thomas-Ulrich) [inline docs/first-model/lua-scripting.rst:46]: please check out the xxx

File: docs/first-model/post-processing.rst [hunk 8]
```diff
@@ -1,6 +1,36 @@
...
 
-   This page is under construction.
++--------------------+------------------------------------------------------------------------+------+
+| Name               | Description                                                            | Unit |
++====================+========================================================================+======+
+| **Time**           | Time at which the output is written                                    | s    |
++--------------------+------------------------------------------------------------------------+------+
+| **state**          | State variable :math:`\phi = f_0 + b \ln{\left(V_0 \theta / L\right)}` |      |
++--------------------+------------------------------------------------------------------------+------+
+| **slip0**          | Slip on the fault                                                      | m    |
```
Comment (Thomas-Ulrich) [inline docs/first-model/post-processing.rst:15]: fault slip

File: docs/first-model/post-processing.rst [hunk 9]
```diff
@@ -1,6 +1,36 @@
...
+-------
+**Fault (probe) outputs (2D)** 
 
-   This page is under construction.
++--------------------+------------------------------------------------------------------------+------+
+| Name               | Description                                                            | Unit |
++====================+========================================================================+======+
+| **Time**           | Time at which the output is written                                    | s    |
++--------------------+------------------------------------------------------------------------+------+
+| **state**          | State variable :math:`\phi = f_0 + b \ln{\left(V_0 \theta / L\right)}` |      |
```
Comment (Thomas-Ulrich) [inline docs/first-model/post-processing.rst:13]: maybe remove the formula as it depends on the friction law?
(or maybe there is only one friction law implemented so far?)

File: docs/first-model/post-processing.rst [hunk 10]
```diff
@@ -1,6 +1,36 @@
...
+| Name               | Description                         | Unit |
++====================+=====================================+======+
+| **Time**           | Time at which the output is written | s    |
++--------------------+-------------------------------------+------+
+| **u**              | Displacement                        | m    |
++--------------------+-------------------------------------+------+
+
+Post-processing tools
+---------------------
+Tools related to the post-processing of the tandem output can be found in `TEAR-ERC/tandem-tools <https://github.com/TEAR-ERC/tandem-tools.git>`_ repository.
```
Comment (Thomas-Ulrich) [inline docs/first-model/post-processing.rst:36]: found in the

File: docs/first-model/run-model.rst (hunk 5 above)
Comment (Thomas-Ulrich) [reply docs/first-model/run-model.rst:4]: What about:
Tandem is invoked with the following syntax

File: docs/first-model/run-model.rst [hunk 11]
```diff
@@ -1,11 +1,25 @@
...
 
-   This page is under construction.
+.. code:: console
+
+   $ ./tandem ../examples/tandem/2d/tutorial.toml --discrete_green yes --petsc -ksp_type preonly -pc_type lu -pc_factor_mat_solver_type mumps -ts_type rk -ts_rk_type 5dp -ts_rtol 1e-8 -ts_atol 1e-50 -ts_adapt_wnormtype infinity -ts_dt 0.0001 -ts_monitor
+
+:code:`--discrete_green yes` is equivalent to adding :code:`mode=QDGreen` in the paramter file.
+
+
+Alternatively, to make the command line neater, multiple PETSc options can be written in a file and loaded using the :code:`-options_file` option.
```
Comment (Thomas-Ulrich) [inline docs/first-model/run-model.rst:14]: As an alternative for a cleaner command line, multiple PETSc options can be specified in a file and loaded using the :code:`--options_file` option.

**Review by Thomas-Ulrich — CHANGES_REQUESTED** [review]
Thank for this PR, very useful.

File: docs/first-model/lua-scripting.rst (hunk 1 above)
Comment (JeenaYun) [reply docs/first-model/lua-scripting.rst:4]: Sure, sorry for making you working on the grammar and typos. I will run them in the grammarly and fix all the typos.

**Comment by JeenaYun** [comment]
Thanks both for checking the typos, and sorry for making you do so. I reflected the suggestions and ran them in grammarly to correct obvious typos. Also corrected the typo related to the CFL parameter in `app/tandem/SEASConfig.cpp:84`.

**Review by Thomas-Ulrich — APPROVED** [review]
LGTM

**Review by baroryan — APPROVED** [review]
Thanks for doing that! It looks good to me

**Pull request merged**